### PR TITLE
Add more fine-grained libpcl-* packages (in addition to ...

### PR DESCRIPTION
**Pull request by mvieth**
...existing libpcl-all and libpcl-all-dev)

No new dependencies are added, the new items just make it possible to be more specific when asking for PCL module libraries. For `libpcl-common` for example, I copied the definition of `libpcl-all` and deleted all distribution packages except for `libpcl-common*`. Most distributions, except for debian and ubuntu, currently don't have individual packages for PCL modules, only one package for all of PCL.

This is a continuation of https://github.com/ros/rosdistro/pull/26448

Please add the following dependency to the rosdep database.

## Package name:

libpcl-common, libpcl-features, libpcl-filters, libpcl-io, libpcl-kdtree, libpcl-segmentation, libpcl-surface

## Package Upstream Source:

https://github.com/PointCloudLibrary/pcl

## Purpose of using this:

The new items make it possible to be more specific when asking for PCL module libraries. Some PCL modules, e.g. visualization, have a lot of dependencies (tens or hundreds of MB). See also https://github.com/ros-perception/perception_pcl/issues/307 for more details. So it would be great if the user is not forced to install dependencies for PCL modules they don't need.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bullseye/libpcl-common1.11
  - https://packages.debian.org/buster/libpcl-common1.9
  - https://packages.debian.org/bullseye/libpcl-features1.11
  - https://packages.debian.org/buster/libpcl-features1.9
  - https://packages.debian.org/bullseye/libpcl-filters1.11
  - https://packages.debian.org/buster/libpcl-filters1.9
  - https://packages.debian.org/bullseye/libpcl-io1.11
  - https://packages.debian.org/buster/libpcl-io1.9
  - https://packages.debian.org/bullseye/libpcl-kdtree1.11
  - https://packages.debian.org/buster/libpcl-kdtree1.9
  - https://packages.debian.org/bullseye/libpcl-segmentation1.11
  - https://packages.debian.org/buster/libpcl-segmentation1.9
  - https://packages.debian.org/bullseye/libpcl-surface1.11
  - https://packages.debian.org/buster/libpcl-surface1.9
- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/bionic/libpcl-common1.8
   - https://packages.ubuntu.com/focal/libpcl-common1.10
   - https://packages.ubuntu.com/jammy/libpcl-common1.12
   - https://packages.ubuntu.com/bionic/libpcl-features1.8
   - https://packages.ubuntu.com/focal/libpcl-features1.10
   - https://packages.ubuntu.com/jammy/libpcl-features1.12
   - https://packages.ubuntu.com/bionic/libpcl-filters1.8
   - https://packages.ubuntu.com/focal/libpcl-filters1.10
   - https://packages.ubuntu.com/jammy/libpcl-filters1.12
   - https://packages.ubuntu.com/bionic/libpcl-io1.8
   - https://packages.ubuntu.com/focal/libpcl-io1.10
   - https://packages.ubuntu.com/jammy/libpcl-io1.12
   - https://packages.ubuntu.com/bionic/libpcl-kdtree1.8
   - https://packages.ubuntu.com/focal/libpcl-kdtree1.10
   - https://packages.ubuntu.com/jammy/libpcl-kdtree1.12
   - https://packages.ubuntu.com/bionic/libpcl-segmentation1.8
   - https://packages.ubuntu.com/focal/libpcl-segmentation1.10
   - https://packages.ubuntu.com/jammy/libpcl-segmentation1.12
   - https://packages.ubuntu.com/bionic/libpcl-surface1.8
   - https://packages.ubuntu.com/focal/libpcl-surface1.10
   - https://packages.ubuntu.com/jammy/libpcl-surface1.12
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/pcl/pcl/
  - https://packages.fedoraproject.org/pkgs/pcl/pcl-tools/
- Arch: https://www.archlinux.org/packages/
  - not available (only in https://aur.archlinux.org/packages/pcl)
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/sci-libs/pcl
- macOS: https://formulae.brew.sh/
  - https://formulae.brew.sh/formula/pcl
- Alpine: https://pkgs.alpinelinux.org/packages
  - https://pkgs.alpinelinux.org/package/edge/testing/x86_64/pcl-libs
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?show=pcl
- openSUSE: https://software.opensuse.org/package/
  - not available (only in https://build.opensuse.org/package/show/science/pcl)

### CODE REVIEW

**Review by tfoote — COMMENTED** [review]
These are a lot of extra rules with strong versioning included which will need to be updated regularly. How much benefit do they provide to call out the individual libraries versus installing the full package? 

And related I see that you've only called out specific versions for ubuntu and debian. Can this same level of granularity not be achieved on the other platforms?

**Comment by mvieth** [comment]
> These are a lot of extra rules with strong versioning included which will need to be updated regularly. How much benefit do they provide to call out the individual libraries versus installing the full package?

The existing rules `libpcl-all` and `libpcl-all-dev` also link to packages with strong versions, so the additional work for these new rules when updating for a new debian/ubuntu release would not be that much IMO.
Some PCL modules, e.g. visualization, have a lot of dependencies (tens or hundreds of MB). See also https://github.com/ros-perception/perception_pcl/issues/307 for more details. So it would be great if the user is not forced to install dependencies for PCL modules they don't need.

> And related I see that you've only called out specific versions for ubuntu and debian. Can this same level of granularity not be achieved on the other platforms?

Except for ubuntu and debian, I am not aware that other platforms offer the PCL modules (common, filters, features, ...) as individual packages. This may change in the future, I don't know.

**Review by methylDragon — CHANGES_REQUESTED** [review]
Packages look like they're missing

**Comment by mvieth** [comment]
> Packages look like they're missing

I assume you are referring to the error messages that package `pcl` can not be found on platforms `arch` and `opensuse`.
However, the package does exist for both:
https://build.opensuse.org/package/show/science/pcl and https://aur.archlinux.org/packages/pcl
I am not sure why the check does not find them?
Please also note that the two existing rules `libpcl-all` and `libpcl-all-dev` already refer to the package `pcl` on `arch` and `opensuse`. So I didn't introduce any new dependencies.

**Comment by nuclearsandwich** [comment]
> Please also note that the two existing rules `libpcl-all` and `libpcl-all-dev` already refer to the package `pcl` on `arch` and `opensuse`. So I didn't introduce any new dependencies.

This indicates that there are regressions in those rules and ideally those regressions would be cleaned up as a general improvement along with these new changes. However, if you limit the changes to dropping the no-longer-valid definitions for these platforms for your newly added keys then that will at least allow review to proceed.

**Comment by mvieth** [comment]
Thanks for the explanation! I removed the package definitions for arch and opensuse for the newly added keys.

**Comment by mvieth** [comment]
@ivanpauno I have updated the description

**Comment by ivanpauno** [comment]
May be a dup of https://github.com/ros/rosdistro/pull/35289, I will double check when reviewing again

**Review by ivanpauno — APPROVED** [review]

File: rosdep/base.yaml [hunk 1]
```diff
@@ -4624,6 +4624,125 @@ libpcl-all-dev:
   rhel: [pcl-devel]
   slackware: [pcl]
   ubuntu: [libpcl-dev]
+libpcl-common:
+  alpine: [pcl-libs]
```
Comment (ivanpauno) [inline rosdep/base.yaml:4628]: could you add `arch: [pcl]`?

```suggestion
  arch: [pcl]
  alpine: [pcl-libs]
```

File: rosdep/base.yaml [hunk 2]
```diff
@@ -4624,6 +4624,125 @@ libpcl-all-dev:
...
+  debian:
+    bullseye: [libpcl-common1.11]
+    buster: [libpcl-common1.9]
+  fedora: [pcl]
+  freebsd: [libpcl]
+  gentoo: [sci-libs/pcl]
+  macports: [libpcl]
+  nixos: [pcl]
+  openembedded: [pcl@meta-ros-common]
```
Comment (ivanpauno) [inline rosdep/base.yaml:4637]:
```suggestion
  openembedded: [pcl@meta-ros-common]
  opensuse: [pcl]
```

**Review by ivanpauno — COMMENTED** [review]
Could you please apply the suggested changes to all the new entries?
Thanks

**Comment by ivanpauno** [comment]
> May be a dup of #35289, I will double check when reviewing again

It's not exactly a dup, but there's some duplication.
I will merge the first one that's ready, the other one will need a rebase after that.

**Comment by ivanpauno** [comment]
https://github.com/ros/rosdistro/pull/35289 is now a superset of this, and it only needs a few minor changes for CI to pass.
So I will close this one.
If for any reason https://github.com/ros/rosdistro/pull/35289 doesn't get merged, please feel free to reopen.

@mvieth thanks for this PR and for your help in https://github.com/ros/rosdistro/pull/35289!!

**Pull request closed without merging**